Deploying to Kinsta Application Hosting (#2083)
withastro/docs: src/pages/en/guides/deploy/kinsta.md

---
title: Deploy your Astro Site to Kinsta Application Hosting
description: How to deploy your Astro site to the web on Kinsta Application Hosting.
layout: ~/layouts/DeployGuideLayout.astro
i18nReady: true
---

You can use [Kinsta Application Hosting](https://kinsta.com/application-hosting/) to host an Astro site on their cloud hosting.

:::tip[Looking for an example?]
Check out [the official Kinsta Application Hosting Starter project for Astro](https://github.com/kinsta/hello-world-astro)!
:::

> ## Configuring your Astro project
> To host your project on **Kinsta Application Hosting**, you need to:
> - make sure your `package.json` has a `name` field.
> - have a script called `build` in your `package.json`. (Your Astro project should already include this.)
> - install the [`serve`](https://www.npmjs.com/package/serve) package and set the `start` script to `serve dist/`.
> 
> Here are the necessary lines in your `package.json` file:
> ```json title="astro.config.mjs" {2,6} ins={12} "serv dist/"
> {
>   "name": "@example/blog",
>   "scripts": {
>     "dev": "astro dev",
>     "start": "serve dist/",
>     "build": "astro build",
>     "preview": "astro preview",
>     "astro": "astro"
>   },
>   "dependencies": {
>     "astro": "^1.6.10",
>     "serve": "^14.0.1"
>   },
> }
> ```

Reviewer: ``````suggestion
## Configuring your Astro project
To host your project on **Kinsta Application Hosting**, you need to:
- Include a `name` field in your `package.json`. (This can be anything, and will not affect your deployment.)
- Include a `build` script in your `package.json`. (Your Astro project should already include this.)
- Install the [`serve`](https://www.npmjs.com/package/serve) package and set the `start` script to `serve dist/`.

Here are the necessary lines in your `package.json` file:
```json title="astro.config.mjs" {2,6} ins={12} "serv dist/"
{
  "name": "anything", // This is required, but the value does not matter.
  "scripts": {
    "dev": "astro dev",
    "start": "serve dist/",
    "build": "astro build",
    "preview": "astro preview",
    "astro": "astro"
  },
  "dependencies": {
    "astro": "^1.6.10",
    "serve": "^14.0.1"
  },
}
```
``````
OK, given `name`-gate, do you think this makes sense/is helpful enough?  😅

Author: Perfect :)

## How to deploy
Once your project's GitHub repository is connected, you can trigger manual deploys to Kinsta Application Hosting in the **MyKinsta Admin Panel**. You can also set up automatic deployments in your admin panel.

### Configuring a new Kinsta application
1. Go to the [My Kinsta](https://my.kinsta.com/) admin panel.

> 2. Go to the **Applications** tab

Reviewer: ```suggestion
2. Go to the **Applications** tab.
```

---
title: Deploy your Astro Site to Kinsta Application Hosting
description: How to deploy your Astro site to the web on Kinsta Application Hosting.
layout: ~/layouts/DeployGuideLayout.astro
i18nReady: true
---

You can use [Kinsta Application Hosting](https://kinsta.com/application-hosting/) to host an Astro site on their cloud hosting.

:::tip[Looking for an example?]
Check out [the official Kinsta Application Hosting Starter project for Astro](https://github.com/kinsta/hello-world-astro)!
:::

> ## Project configuration

Reviewer: ```suggestion
## Configuring your Astro project
```

To host your project on **Kinsta Application Hosting**, you need to:

> - make sure your `package.json` has a `name` field.

Reviewer: Can you describe what should/shouldn't go here? I notice that in your example it's not formatted like a domain, and it starts with an `@` and also seems to have a subdirectory. I can't tell from this what the proper format has to be, and whether these things are all requirements. Can you provide further guidance on this for the reader?

Author: For some reason, Buildpacks (the tech behind building everything on Kinsta) requires to have the "name" field. It's required and that's it. You could add any text there.

It's quite important because sometimes there are package.json files without this field.

Reviewer: interesting! So, it's "used" somehow by Buildpacks, but not in any way that would affect your projet? It could literally be *anything*?

Author: You can enter "qwerty," and it will be OK. It's weird, but it is as it is.

Reviewer: I'll take it! Just need to know how to describe it. lol... not my place to judge... 😄

Author: I think the later commit about the package name solves everything, and we can close this.

> - have a script called `build` in your `package.json.`

Reviewer: ```suggestion
- have a script called `build` in your `package.json`. (Your Astro project should already include this.)
```

> - install the `serve` package and set the `start` script to `serve dist/`.

Reviewer: ```suggestion
- install the [`serve`](https://www.npmjs.com/package/serve) package and set the `start` script to `serve dist/`.
```

> This is an example of a correct `package.json` file:

Reviewer: ```suggestion
Here are the necessary lines in your `package.json` file:
```

> ```json
> {
>   "name": "@example/blog",
>   "version": "0.0.1",
>   "private": true,
>   "scripts": {
>     "dev": "astro dev",
>     "start": "serve dist/",
>     "build": "astro build",
>     "preview": "astro preview",
>     "astro": "astro"
>   },
>   "dependencies": {
>     "astro": "^1.1.7",
>     "@astrojs/mdx": "^0.11.1",
>     "@astrojs/rss": "^1.0.0",
>     "@astrojs/sitemap": "^1.0.0",
>     "serve": "^14.0.1"
>   },
>   "devDependencies": {
>     "astro": "^1.1.7",
>     "@astrojs/renderer-preact": "^0.5.0"
>   }
> }
> ```

Reviewer: ``````suggestion
```json title="astro.config.mjs" {2,6} ins={12} "serv dist/"
{
  "name": "@example/blog",
  "scripts": {
    "dev": "astro dev",
    "start": "serve dist/",
    "build": "astro build",
    "preview": "astro preview",
    "astro": "astro"
  },
  "dependencies": {
    "astro": "^1.6.10",
    "serve": "^14.0.1"
  },
}
```
``````
Unless those other dependencies are required, I would not show them here. The version of Astro (and others) was also very old, so it is better to show as little as possible. I also highlighted the contents.

## How to deploy

> Your deployment to Kinsta Application Hosting will happen by pressing the button inside of **MyKinsta Admin Panel** or by setting automatic deployments.

Reviewer: ```suggestion
Once your project's GitHub repository is connected, you can trigger manual deploys to Kinsta Application Hosting in the **MyKinsta Admin Panel**. You can also set up automatic deployments in your admin panel.
```

> ### Website Ui Deployment

Reviewer: ```suggestion
### Configuring a new Kinsta application
```

> 1. Go to [My Kinsta](https://my.kinsta.com/) panel.

Reviewer: ```suggestion
1. Go to the [My Kinsta](https://my.kinsta.com/) admin panel.
```
